pandas/core/missing.py

```diff
@@ -165,6 +165,16 @@ def interpolate_1d(xvalues, yvalues, method='linear', limit=None,
     elif limit < 1:
         raise ValueError('Limit must be greater than 0')
 
+    if (maxgap is not None) and (limit is not None):
+        raise ValueError('maxgap cannot be used together with limit')
```

Comment:
Is this a fundamental limitation, or just hasn't been implemented yet? I could imagine using both of these at the same time.

If it just hasn't been implemented yet, perhaps raise a NotImplementedError

Reply:
To me, it does not make sense to use both, `maxgap` and `limit`, at the same time, since I want to use `maxgap` to avoid the behavior of `interplotate(limit=N). The purpose of `maxgap` is that NaN-gaps that are too long are not filled in any way. The reasoning is that it can be justified to interpolate, e.g. short periods without sensor data, but it does not make sense to interpolate only some values into a longer gap. 

I could imagine a theoretical use case that somebody would want to interpolate at least some values up to `limit` in a NaN-gap that is longer than `maxgap`. But i do not see why anybody would want to do that with real data.

From the implementation, I think, `limit` and `maxgap` could be combined. We just have to combine their `preserve_nans` indices. I would have to change the logic of the if-statement a little.